design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls

```diff
@@ -76,9 +76,24 @@ type AddNegotiableQuoteItemsOutput {
 
 input SetNegotiableQuoteShippingAddressInput {
     quote_uid: ID! @doc(description: "ID obtained from NegotiableQuote type")
-    customer_address_id: Int! @doc(
+    customer_address_id: Int @doc(
         description: "ID obtained from the CustomerAddress type. A new address can be added using Mutation.createCustomerAddress"
     )
+    address: NegotiableQuoteAddressInput
+}
+
+input NegotiableQuoteAddressInput {
```

Comment:
I would suggest making a `QuoteAddressInputInterface` in QuoteGraphQL with these fields and have both `CartAddressInput` and `NegotiableQuoteAddressInput` implement it.

Reply:
I thought we were holding off on all unification via interfaces for now? Would it not be inconsistent to do that for the CartAddressInput and not the Cart itself? 

Also, in all likelihood, this wouldn't cause any issues with any existing CartAddress functionality. However, isn't there an added element of risk?

Reply:
It's not inconsistent.  It's not changing anything about what shows up on `CartAddressInput`, but it lets us use the same backing type for Negotiable Quotes.  Everything in the schema shows up and is treated exactly the same, just like moving a function to a parent class in code.  There's no risk here, it's not renaming anything that's already there or changing where it's already accessible or how it's extended.

We're holding off on unifying the existing schema because we need to investigate the level of effort involved, not because it isn't the right thing to do.  This is a new addition to the schema; not doing it with a unified interface now just means that much more needs to happen in the future.

Reply:
It also doesn't currently make a lot of sense to do a unified interface between `Cart` and `NegotiableQuote`.  The only field they share with an identical definition is `prices`.

Reply:
> The only field they share with an identical definition is prices.

There's also `items`. In general though, I think the unified interface was in anticipation for checkout where there will be several more common fields. Still, it's better to hold off.

Reply:
> There's no risk here, it's not renaming anything that's already there or changing where it's already accessible or how it's extended.
As long as there's no risk, then I'm in favor of just a `QuoteAddressInputInterface`. What are your thoughts on a `NegotiableQuoteAddressInput` though?

Reply:
I prefer `QuoteAddressInput` myself as ideally that's what it should have been in both places, but since we have to have them separate I'm fine with `NegotiableQuoteAddressInput` to match the naming in the rest of this module.